### ISSUE TITLE
Custom Loss function or MAE

### ISSUE THREAD

**Issue by JoshuaC3** [baidu/fast_rgf#10]
Is it possible to use MAE or a custom loss function as the objective/loss function for this?

Examples of other tree based learners that do this or something similar are:
https://github.com/dmlc/xgboost/blob/master/demo/guide-python/custom_objective.py
https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.md#core-parameters
http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html
